File: neet/arrays/twosum.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
		unordered_map<int, int> comp;
		vector<int> ans;
		int it = 0;
		for (int el: nums) {
			int diff = target - nums[it];
			if (comp.count(diff) >0) {
				ans.push_back(it);
				ans.push_back(comp[diff]);
				return ans;
			}
			comp.insert(make_pair(el, it));
			it++;
		}
		return ans;
    }
};

File: neet/arrays/twosum.py
#!/usr/bin/env python3

class Solution:
	def twoSum(self, nums: List[int], target: int) -> List[int]:
		d = {}
		for i, num in enumerate(nums):
			c = target - num
			if c in d and d[c] != i:
				return [i, d[c]]

			d[num] = i


